File: reduks/src/main/java/com/beyondeye/reduks/Thunk.java
package com.beyondeye.reduks;

/**
 *  keep this interface in java to leverage kotlin SAM conversion (lambda automatical converted to implementation of java interface)
 *
 * Created by daely on 5/24/2016.
 */
public interface Thunk<S> extends Action {
    Object execute(NextDispatcher dispatcher, S state);
}

File: reduks/src/main/kotlin/com/beyondeye/reduks/Thunk.kt
package com.beyondeye.reduks

/**
 * single method interface, mainly used because kotlin does not support yet type alias for function types
 * Created by daely on 5/24/2016.
 */
interface Thunk<S> : Action {
    fun execute(dispatcher: (Any)->Any, state: S): Any
}
